modules/apps/application-list/application-list-my-account-permissions/src/main/java/com/liferay/application/list/my/account/permissions/internal/PanelAppMyAccountPermissions.java

```diff
@@ -168,10 +168,10 @@ private void _initPermissions(
 	@Reference
 	private RoleLocalService _roleLocalService;
 
-	private ServiceTracker<PanelApp, PanelApp> _serviceTracker;
+	private ServiceTrackerMap<String, PanelApp> _serviceTrackerMap;
```

Comment:
@MrJohn1911 Any reason to change the service tracker to service tracker map here?

- The map here uses an `Eager` customizer, so it will not be lazy;
- No code queries from the map.

cc @rafaprax 

This PR https://github.com/liferay-core-infra/liferay-portal/pull/5823 also did this.

Reply:
Hi @dantewang, there is no specific reason to use this interface, we just did this because it was a pattern that we were following for the service trackers that we found in our refactors.

Reply:
@MrJohn1911 @rafaprax 

After discussion with Shuyang, I'll push this change back. Please resend.

See:

> EagerServiceTrackerCustomizer should be used with extra caution. Most of existing ServiceTracker and EagerServiceTrackerCustomizer usages should be refactored to avoid eager service piping or pre-processing logic. But that can be very complex to do (although technically speaking, they are all doable).
>
>if the converting from ServiceTracker to ServiceTrackerMap with EagerServiceTrackerCustomizer only does a structural change, then no need to do it.

Basically, this is only a structural change. We can't achieve lazy opening here because of `EagerServiceTrackerCustomizer`; we don't need the extra logic that constructs the map, either.